widget/gridlayout.go

```diff
@@ -137,45 +137,39 @@ func (g *GridLayout) Layout(widgets []PreferredSizeLocateableWidget, rect image.
 	firstStretchedCol, firstStretchedRow := true, true
 	for _, w := range widgets {
 		cw := colWidths[c]
-		if w.GetWidget().Visibility == Visibility_Hide {
-			c++
-			continue
-		} else if w.GetWidget().Visibility == Visibility_Hide {
-			c++
-			x += cw + g.columnSpacing
-			continue
-		}
-
-		if g.columnStretched(c) {
-			cw = stretchedColWidth
-			if firstStretchedCol {
-				cw = firstStretchedColWidth
-				firstStretchedCol = false
+		ch := rowHeights[r]
+		if w.GetWidget().Visibility == Visibility_Show {
+			if g.columnStretched(c) {
+				cw = stretchedColWidth
+				if firstStretchedCol {
+					cw = firstStretchedColWidth
+					firstStretchedCol = false
+				}
 			}
-		}
 
-		ch := rowHeights[r]
-		if g.rowStretched(r) {
-			ch = stretchedRowHeight
-			if firstStretchedRow {
-				ch = firstStretchedRowHeight
-				firstStretchedRow = false
+			if g.rowStretched(r) {
+				ch = stretchedRowHeight
+				if firstStretchedRow {
+					ch = firstStretchedRowHeight
+					firstStretchedRow = false
+				}
 			}
-		}
 
-		ww, wh := cw, ch
-		wx, wy := x, y
+			ww, wh := cw, ch
+			wx, wy := x, y
 
-		ld := w.GetWidget().LayoutData
-		if gld, ok := ld.(GridLayoutData); ok {
-			wx, wy, ww, wh = g.applyLayoutData(gld, wx, wy, ww, wh, x, y, cw, ch)
-		}
+			ld := w.GetWidget().LayoutData
+			if gld, ok := ld.(GridLayoutData); ok {
+				wx, wy, ww, wh = g.applyLayoutData(gld, wx, wy, ww, wh, x, y, cw, ch)
+			}
 
-		w.SetLocation(image.Rect(rect.Min.X+wx, rect.Min.Y+wy, rect.Min.X+wx+ww, rect.Min.Y+wy+wh))
+			w.SetLocation(image.Rect(rect.Min.X+wx, rect.Min.Y+wy, rect.Min.X+wx+ww, rect.Min.Y+wy+wh))
 
+		}
 		c++
-		x += cw + g.columnSpacing
-
+		if w.GetWidget().Visibility != Visibility_Hide {
+			x += cw + g.columnSpacing
```

Comment:
I’m not entirely sure this here is correct. I guess if a grid has 2 columns and 2 rows and the first widget at the top left is `Visibility_Hide`, perhaps it depends on the visibility of the one at the bottom left to know if we add the column spacing or not?

Reply:
I think the main problem is the if statement near the begining is wrong in the old code. It is essentially saying if its "Hide", just continue, otherwise, if it is Visibility_Hide_Blocking include the column width.

		if w.GetWidget().Visibility == Visibility_Hide {
			c++
			continue
		} else if w.GetWidget().Visibility == Visibility_Hide **_Blocking** {
			c++
			x += cw + g.columnSpacing
			continue
		}

Reply:
To be honest, I am not 100% certain of the original code here, so will have to play with what you have a bit.